### ISSUE TITLE
[R-package] lgb.cv returns negative best score with an absolute value metric

### ISSUE THREAD

**Issue by Laurae2** [microsoft/LightGBM#2158]
It seems the R package currently is able to return a negative value even when the metric is never negative, due to rounding issues. Happens especially on lgb.cv, but sometimes happens on lgb.train.

```r
library(lightgbm)
data(agaricus.train, package = "lightgbm")
train <- agaricus.train
dtrain <- lgb.Dataset(train$data, label = train$label)
data(agaricus.test, package = "lightgbm")
test <- agaricus.test
dtest <- lgb.Dataset.create.valid(dtrain, test$data, label = test$label)
params <- list(objective = "regression", metric = "l2")
valids <- list(test = dtest)
model <- lgb.train(params,
                   dtrain,
                   10,
                   valids,
                   min_data = 1,
                   learning_rate = 1,
                   early_stopping_rounds = 10)
```

best_score is: -6.44165e-17
